pkg/controller/deployment/deployment_controller_test.go

```diff
@@ -150,28 +159,43 @@ type fixture struct {
 
 	// Actions expected to happen on the client. Objects from here are also
 	// preloaded into NewSimpleFake.
-	actions []core.Action
-	objects []runtime.Object
+	actions     []core.Action
+	objectTests []func(runtime.Object)
```

Comment:
it looks like this is all about comparing the object.

Maybe the fixture can simply have a boolean to validate the equality of objects. You can use `cmpopts` to ignore certain fields.

Reply:
I also need to add tests verifying correct fenceposts in a variety of surge conditions